### PR TITLE
Lzhan55/ase plugin2

### PR DESCRIPTION
**Pull request by lzhan55**
* ASE is implemented as a OPAE plugin
* Added an ASE config file to run ASE only
* Samples can be linked to libopae-c.so to run ASE. Previously it was linked to libopae-c-ase.so.

### CODE REVIEW

**Comment by michael-adler** [comment]
How does the plugin mgr decide whether to choose ASE or xfpga?

**Comment by lzhan55** [comment]
@michael-adler You can use opae.cfg to enable ASE if you only want ASE to run. When both xfpga and ASE are enabled in the opae.cfg file, both plugin libraries will be loaded. And the application can do fpgaEnumerate() with ASE's GUID to run ASE. Or, the application can run both xfpga and ASE concurrently.

**Comment by michael-adler** [comment]
What is your strategy for keeping existing scripts/Makefiles working when they specify -lopae-c-ase?

Have you thought about what to do with the "with_ase" script? Perhaps it could do something special and force the plugin manager to pick ASE.

I think we should avoid two things (perhaps you've done this already):
1. Don't force developers to edit opae.cfg every time that want to switch between FPGA and ASE.
2. Don't force developers to modify their software in order to use ASE.

Emphases on the word "force". If someone wants to use fpgaEnumerate() to select between simulation and hardware using your new scheme that's fine. Just don't make it a requirement.

File: libopae/pluginmgr.c [hunk 1]
```diff
@@ -231,6 +231,7 @@ STATIC int opae_plugin_mgr_initialize_all(void)
 	int res;
 	opae_api_adapter_table *aptr;
 	int errors = 0;
+	int count = 0;
```
[the remark this answers is not in the document]
Comment (lzhan55) [reply libopae/pluginmgr.c]: Will remove it.

**Comment by tswhison** [comment]
Please do address Michael's comments before merging.

**Comment by lzhan55** [comment]
@michael-adler  We can create a symbol link of libopae-c-ase.so to libopae-c.so. And we can check whether LD_PRELOAD="libopae-c-ase.so" was set to let pluginmgr.c to load the ASE plugin correspondingly. Let's set up a meeting to discuss its implementation details.

**Pull request merged**